java-manual/modules/ROOT/pages/reactive.adoc

```diff
@@ -1,18 +1,24 @@
 = Control results flow with reactive streams
 
+[NOTE]
+====
+The Reactive API is recommended for applications that already work in a reactive programming style, and which have needs that only Reactive workflows can address.
+For all other cases, the xref:query-simple.adoc[sync] and xref:async.adoc[async] APIs are recommended.
```

Comment:
Maybe noting that the aync api works quite well with virtual threads on Java 21+, otherwise I feel it's close to be as hard as reactive, not quite, but still.